lib/isZodErrorLike.test.ts

```diff
@@ -0,0 +1,48 @@
+import * as zod from 'zod';
+import { isZodErrorLike } from './isZodErrorLike.ts';
+
+class CustomZodError extends Error {
+  issues: zod.ZodIssue[];
+
+  constructor(message: string) {
+    super(message);
+    this.name = 'ZodError';
+    this.issues = [];
+  }
+}
+
+describe('isZodErrorLike()', () => {
+  test('returns true when argument is an actual instance of ZodError', () => {
+    const err = new zod.ZodError([
+      {
+        code: zod.ZodIssueCode.custom,
+        path: [],
+        message: 'foobar',
+        fatal: true,
+      },
+    ]);
+
+    expect(isZodErrorLike(err)).toEqual(true);
+  });
+
+  test('returns true when argument resembles a ZodError', () => {
+    const err = new CustomZodError('foobar');
+
+    expect(isZodErrorLike(err)).toEqual(true);
+  });
+
+  test('returns false when argument is generic Error', () => {
+    expect(isZodErrorLike(new Error('foobar'))).toEqual(false);
+  });
+
+  test('returns false when argument is not an Error instance', () => {
+    expect(isZodErrorLike('foobar')).toEqual(false);
```

Comment:
nit: I would have those assertions separately. Also, I would test for generic `Error` instances, or even `ZodError` instances with no `issues`.

Reply:
Fixed [0102e6e](https://github.com/causaly/zod-validation-error/pull/304/commits/0102e6e5b3c789bb47000d8309fd86c253d42646)